File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        resMap = {}

        for i in range(len(nums)):
            j = target - nums[i]
            if j in resMap:
                return [i, resMap[j]]
            resMap[nums[i]] = i

        return []

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int result[] = new int[2];
        Map<Integer, Integer> resMap = new HashMap<>();

        for(int i=0;i<nums.length;i++){
            
            int j = target - nums[i];
            if(resMap.containsKey(j)){
                result[0] = i;
                result[1] = resMap.get(j);
            }
            resMap.put(nums[i],i);
        }
        return result;
    }
}